language/move-prover/docgen/tests/sources/different_visbilities.move

```diff
@@ -0,0 +1,15 @@
+address 0x2 {
+module TestViz {
+    /// This is a public function
+    public fun this_is_a_public_fun() { }
+
+    // /// This is a public friend function
+    // public(friend) fun this_is_a_public_friend_fun() {}
+
+    /// This is a public script function
+    public(script) fun this_is_a_public_script_fun() {}
```

Comment:
what is difference between `public(script) fun` and `public fun`?

Reply:
In general `public(script)` functions are functions that can only be called by transactions (or by other `public(script) fun`s). As an intuition, a `public(script) fun` can be viewed as defining an RPC endpoint to the blockchain for clients.  

The way I think of it is we have two APIs that we're defining in a module with `public(script) fun` and `public fun`
* `public fun` describes the on-chain API of the module -- what functions can other modules call
* `public(script) fun` describes the off-chain to on-chain API -- what operations can clients perform, and what transactions can they send. So, instead of having the (Move) transaction scripts that we have today, e.g., `peer_to_peer_with_metadata`,  this would instead be defined as a `public(script) fun peer_to_peer_with_metadata(...)` in a module `M`. Transactions could then be sent as essentially "call the `peer_to_peer_with_metadata` function in `M` with these arguments" instead of passing the actual bytes of the transaction script to be executed like it is today.